esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -4192,46 +4213,47 @@ int CWsWuFileHelper::zipAFolder(const char* folder, bool gzip, const char* zipFi
 }
 
 void CWsWuFileHelper::createWUZAPFile(IEspContext& context, IConstWorkUnit* cwu, CWsWuZAPInfoReq& request,
-    StringBuffer& zipFileName, StringBuffer& zipFileNameWithPath, unsigned _thorSlaveLogThreadPoolSize)
+    StringBuffer& zapFileName, StringBuffer& zipFileNameWithFullPath, unsigned _thorSlaveLogThreadPoolSize)
 {
-    StringBuffer zapReportNameStr, folderToZIP, inFileNamePrefixWithPath;
-    Owned<IFile> zipDir = createWorkingFolder(context, request.wuid.str(), "ZAPReport_", zapReportNameStr, folderToZIP);
-    setZAPFile(request.zapFileName.str(), zapReportNameStr.str(), zipFileName, zipFileNameWithPath);
+    setZAPReportName(request, cwu, zapFileName);
+
+    Owned<IFile> tempDir = createUniqueTempDirectory();
+    const char* tempDirName = tempDir->queryFilename();
+    setZIPFileName(tempDirName, zipFileNameWithFullPath);
```

Comment:
I think this is should be needed, can be simplified to:
    zipFileNameWithFullPath.append(zipFileNameWithFullPath).append(PATHSEPCHAR).append("zapreport.zip");

(as noted before the temporary zip filename is arbitrary, it is not the same as the browser / client filename used that the user sees)

It would also be better to be a file in the tmp directory, so that a single call to recusively delete this temp directory, clear it at same time, but that would also mean you would need to create (and delete) it outside of this function, when it is called.

Reply:
I guess I should replace the setZIPFileName() with:
zipFileNameWithFullPath.set(tempDirName).append(PATHSEPCHAR).append("zapreport.zip");

And move it (and line 4220/4221) outside of this function since the zapreport.zip should be deleted after it is sent.